packages/pymedphys_dicom/src/pymedphys_dicom/ct/extend.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2019 Cancer Care Associates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version (the "AGPL-3.0+").
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+
+import os
+from copy import deepcopy
+from glob import glob
+import datetime
+import random
+from collections import deque
+
+import pydicom
+from ..constants.uuid import PYMEDPHYS_ROOT_UID
+
+
+def extend(input_dir, output_dir, index_to_copy, number_of_slices):
+    filepaths = glob(os.path.join(input_dir, '*'))
+    dicom_datasets = load_dicom_into_deque(filepaths)
+    extend_datasets(dicom_datasets, index_to_copy, number_of_slices)
+
+    filenames = [
+        os.path.basename(filepath)
+        for filepath in filepaths
+    ]
+
+    common_prefix = os.path.commonprefix(filenames)
+
+    number_of_digits = len(str(len(dicom_datasets)))
+    new_filenames = [
+        "{}{}.dcm".format(
```

Comment:
Not `CT.SOPInstanceUID`? That's how both our old Philips and new Toshiba CT scanners name CT files...

It's actually pretty standard across all DICOM modalities, not just CT.

Reply:
I'd certainly be happy to change to that. This was just based on how Monaco exports the DICOM. More than happy to use a different (better) standard :).

Reply:
So I changed it to `<Whatever prefix was common before><SOPInstanceUID>.dcm`

Reply:
I don't think that works... what if part of the common prefix includes part of the SOPInstanceUID. I think it always will, actually...

Reply:
> This was just based on how Monaco exports the DICOM.

Ah. Sorry, I thought only Varian were annoying enough to break convention...

> More than happy to use a different (better) standard

Not sure it's better, to be honest. Often the alphabetical order of conventionally named CT slices named doesn't match slice order. Maybe your way is better after all.

I only raised it because we adopted that convention in the anonymiser. I.e. a file basename of:

`<2-char-modality>.<SOPInstanceUID>.dcm`

Reply:
Ahh okay, I do like it printing in slice order. Alright, I might revert it...

Reply:
No worries :)

Reply:
I'll leave this one unresolved so I remember to revert :)